lib/modules.nix

```
@@ -366,11 +372,14 @@ rec {
       };
     defsFinal = defsFinal'.values;
 
-    # Type-check the remaining definitions, and merge them.
-    mergedValue = foldl' (res: def:
-      if type.check def.value then res
-      else throw "The option value `${showOption loc}' in `${def.file}' is not of type `${type.description}'.")
-      (type.merge loc defsFinal) defsFinal;
+    # Type-check the remaining definitions, and merge them. Or throw if no definitions.
+    mergedValue =
+      if isDefined
+      then foldl' (res: def:
+        if type.check def.value then res
+        else throw "The option value `${showOption loc}' in `${def.file}' is not of type `${type.description}'.")
+        (type.merge loc defsFinal) defsFinal
+      else throw "The option `${showOption loc}' is used but not defined.";
```

This could ask `type` for a placeholder value; something like

```suggestion
      else t.placeholder or (throw "The option `${showOption loc}' is used but not defined.");
```

`nullOr` can then have `{ placeholder = null; }`.

Not too complicated I think, while giving anyone the opportunity to add placeholders as appropriate for their use cases.

---

Do not forget to add the synchronization comment from `valueDefined`.

---

Done that